refactor(forms): remove deprecated uses from the unit tests

JeanMeche opened the pull request:
Jasmine has deprecated the `expectationFailOutput` argument and replaced it by the `withContext()` method

Also removing all references to #24571 from the forms unit tests as the non null assertions are fine in the context.

## PR Type
What kind of change does this PR introduce?


- [x] Refactoring (no functional changes, no api changes)


## Does this PR introduce a breaking change?


- [x] No

JeanMeche committed:
refactor(forms): remove deprecated uses from the unit tests

Jasmine has deprecated the `expectationFailOutput` argument and replaced it by the `withContext()` method

Also removing all references to #24571 from the forms unit tests as the non null assertions are fine in the context.

AndrewKushnir commented on packages/forms/test/template_integration_spec.ts:
```diff
@@ -773,12 +805,18 @@ import {NgModelCustomComp, NgModelCustomWrapper} from './value_accessor_integrat
+                 .withContext('Expected form value to update on programmatic change.')
                  .toEqual(
-                     {name: 'Carson'}, 'Expected form value to update on programmatic change.');
+                     {name: 'Carson'},
```
```suggestion
                     {name: 'Carson'}
```

AndrewKushnir commented on packages/forms/test/template_integration_spec.ts:
```diff
@@ -773,12 +805,18 @@ import {NgModelCustomComp, NgModelCustomWrapper} from './value_accessor_integrat
-                 .toBe(false, 'Expected validation to run immediately on programmatic change.');
+                 .withContext('Expected validation to run immediately on programmatic change.')
+                 .toBe(
+                     false,
```
```suggestion
                     false
```

AndrewKushnir reviewed (commented):
@JeanMeche thanks for the refactoring 👍
The change looks good, I've just left a couple minor comments.

JeanMeche committed:
refactor: remove todos on forms tests

Removing the todos implied using fakeAsync and passing an async validator as async (3rd parameter).

AndrewKushnir reviewed (approved):
@JeanMeche I've re-reviewed this PR, the changes are looking good 👍 

Please don't push more commits into this branch, since it'd trigger the re-review process (thus would delay the merge). If there are some extra changes, please create a new branch and a new PR based on it.

Thank you.

dylhunn commented:
This PR was merged into the repository by commit 33b9cb8693288f0c240259da461b0cf3f1412214.
